### DISCUSSION TITLE
Connect keycloak container to PostgreSQL

### DISCUSSION THREAD

**Discussion by koplas** [dasniko/testcontainers-keycloak#161] [Q&A]
Hi,
are there any examples of how to connect a keycloak testcontainer to PostgreSQL? The default H2 database does not support all PostgreSQL features used in the extension I  need to test.

Thanks!

**Comment by dasniko** [comment]
I didn't add an api for this, as this would, from my experience with similar things, yield in too much service requests and issues b/c most people wouldn't understand how to use it properly.

But you can still configure the container like you would do with a regular Keycloak container (actually it _is_ a regular KC container) with environment variables (`.withEnv(...)`) and/or cli arguments (`.withCustomCommand(...)` - this is not documented, see source code if needed).
